runtime/error.c

```diff
@@ -181,3 +194,115 @@ TaggedPtr BAL_LANG_ERROR_NAME(message)(TaggedPtr error) {
     ErrorPtr ep = taggedToPtr(error);
     return ep->message;
 }
+
+void _bal_print_mangled_name(const char *mangledName, FILE *fp) {
+    const char *localName;
+    DemangleResult res = demangle(mangledName, &localName, NULL);
+
+    if (res == NON_PUBLIC_BALLERINA_NAME) {
+        fprintf(fp, "%s", localName);
+    }
+    else if (res == PUBLIC_BALLERINA_NAME) {
+        fprintf(fp, "%s (", localName);
+        demangle(mangledName, &localName, fp);
+        fputs(")", fp);
+    }
+    else if (res == NON_BALLERINA_NAME || res == _B_NON_BALLERINA_NAME) {
+        fprintf(fp, "%s", mangledName);
+    }
+}
+
+static DemangleResult demangle(const char *mangledName, const char **localName, FILE *fp) {
+    if (mangledName[0] != '_' || mangledName[1] != 'B') {
+        return NON_BALLERINA_NAME;
+    }
+    mangledName += 2;
+    if (mangledName[0] == '_') {
+        *localName = mangledName + 1;
+        return NON_PUBLIC_BALLERINA_NAME;
+    }
+
+    bool ballerinaOrg = false; 
+    if (mangledName[0] == 'b') {
+        ballerinaOrg = true;
+        mangledName++;
+    }
+
+    const char *org;
+    if (mangledName[0] == '0') {
+        mangledName++;
+        org = NULL;
+    }
+    else {
+        if (!demangleCountedName(&mangledName, &org)) {
+            return _B_NON_BALLERINA_NAME;
+        }
+    }
+    if (fp != NULL) {
+        bool haveOrg = false;
+        if (ballerinaOrg) {
+            fputs("ballerina", fp);
+            haveOrg = true;
+        }
+        if (org != NULL) {
+            fwrite(org, 1, mangledName - org, fp);
+            haveOrg = true;
+        }
+        if (haveOrg) {
+            putc('/', fp);
+        }
+    }
+
+    if (!demangleModules(&mangledName, fp)) {
+        return NON_BALLERINA_NAME;
+    }
+    const char *lastModule;
+    if (!demangleCountedName(&mangledName, &lastModule)) {
+        return NON_BALLERINA_NAME;
+    } 
+    *localName = mangledName;
+    if (fp != NULL) {
+        fwrite(lastModule, 1, mangledName - lastModule, fp);
+    }
+    return PUBLIC_BALLERINA_NAME;
+}
+
+static bool demangleModules(const char **mangledModules, FILE *fp) {
+    if (*mangledModules[0] != 'm') {
+        return true;
+    }
+    *mangledModules = *mangledModules + 1;
+    const char *module;
+    if (!demangleCountedName(mangledModules, &module)) {
+        return false;
+    }
+    if (fp != NULL) {
+        fwrite(module, 1, *mangledModules - module, fp);
+        fputs(".", fp);
+    }
+    return demangleModules(mangledModules, fp);
+}
+
+static bool demangleCountedName(const char **pp, const char **name) {
+    long nChars;
+    int nRead;
+    if (sscanf(*pp, "%ld%n", &nChars, &nRead) <= 0) {
+        return false;
+    }
+    if (nRead > MAX_DEMANGLING_DECIMAL_LENGTH) {
```

Comment:
Why do we need this?

Reply:
I did this, because there was a potential integer overflow as mentioned above. Now we are not doing an addition like `nRead + nChars` we can remove this.